cl/alerts/tasks.py

```diff
@@ -28,117 +30,142 @@ def enqueue_docket_alert(d_pk: int) -> bool:
-            # for this case-user. Ignore it if user doesn't exist in our DB.
-            try:
-                user_profile = UserProfile.objects.get(
-                    recap_email=email_address
-                )
-            except UserProfile.DoesNotExist:
-                # TODO send an email to admins informing the issue
-                continue
-            email_address = user_profile.user.email
-            docket_alert_exist = DocketAlert.objects.filter(
-                docket=d, user=user_profile.user
-            ).exists()
-            if docket_alert_exist:
-                # If a docket alert exists for this @recap.email user, avoid
-                # sending the first email
-                continue
-            first_email = True
-            if user_profile.auto_subscribe:
-                # First time recipient, auto_subscribe True
-                auto_subscribe = True
-                docket_alert = DocketAlert.objects.create(
-                    docket=d, user=user_profile.user
-                )
-            else:
-                # First time recipient, auto_subscribe False
-                auto_subscribe = False
-                docket_alert = DocketAlert.objects.create(
-                    docket=d,
-                    user=user_profile.user,
-                    alert_type=DocketAlert.UNSUBSCRIPTION,
-                )
-        else:
-            # Options for users that already have an active subscription for
-            # this case
-            first_email = auto_subscribe = False
-            user = User.objects.filter(
-                email=email_address, docket_alerts__docket=d
-            ).first()
-            docket_alert = DocketAlert.objects.get(docket=d, user=user)
-
-        subject_context["first_email"] = first_email
-        subject_context["auto_subscribe"] = auto_subscribe
-        email_context["docket_alert"] = docket_alert
-        email_context["first_email"] = first_email
-        email_context["auto_subscribe"] = auto_subscribe
+    messages = []
+    for recipient_tuple in recipients:
+        email_context["docket_alert_secret_key"] = recipient_tuple[1]
         subject = subject_template.render(subject_context).strip()  # Remove
         # newlines that editors can insist on adding.
         msg = EmailMultiAlternatives(
             subject=subject,
             body=txt_template.render(email_context),
             from_email=settings.DEFAULT_ALERTS_EMAIL,
-            to=[email_address],
+            to=[recipient_tuple[0]],
             headers={f"X-Entity-Ref-ID": f"docket.alert:{d.pk}"},
         )
         html = html_template.render(email_context)
         msg.attach_alternative(html, "text/html")
         messages.append(msg)
-
     return messages
 
 
 # Ignore the result or else we'll use a lot of memory.
 @app.task(ignore_result=True)
-def send_docket_alert(
+def send_alert_and_webhook(
     d_pk: int,
     since: datetime,
     recap_email_recipients: list[str] = [],
```

Comment:
I guess I missed this before, but `[]` is mutable and will cause all kinds of fun problems, probably. Instead this should be `= None`, and then you should add a something like:

```
if recap_email_recipients is None:
    recap_email_recipients = []
```

Reply:
Done, I've chanted it to `= None` and applied the suggestion:
```
if recap_email_recipients is None:
    recap_email_recipients = []
```
Thanks.